tests/intentManager.js

```diff
@@ -17,17 +18,48 @@ define([
     widgets,
     INTENT_WITH_UNKNOWN_ATTRS_XML,
     INTENT_WITH_NO_MUG_XML,
-    PRINTING_INTENT_XML
+    PRINTING_INTENT_XML,
+    CUSTOM_INTENT_XML
 ) {
     var assert = chai.assert,
-        call = util.call;
+        call = util.call,
+        templates =  [
+            {
+                icon: "icon-map-marker",
+                name: "Area Mapper",
+                id: "com.richard.lu.areamapper",
+                extra: {ext: "value"},
```

Comment:
Those were moved from later in the file.

Those can actually auto fill in the extra/response fields based on template selection (can be done in vellum, but not present in HQ, tested here https://github.com/dimagi/Vellum/blob/6aed8e75985678776567657a06a9c64f6fa2e15d/tests/intentManager.js#L158-L159)